src/dotnet/Program.cs

```diff
@@ -113,6 +113,16 @@ private static int ProcessArgs(string[] args)
 
             var appArgs = (lastArg + 1) >= args.Length ? Enumerable.Empty<string>() : args.Skip(lastArg + 1).ToArray();
 
+            if (verbose.HasValue)
+            {
+                Environment.SetEnvironmentVariable(CommandContext.Variables.Verbose, verbose.ToString());
+            }
+            //This process should inherit AnsiPassThrough or use the default of false.
+            CommandContext.InitializeAnsiPassThroughFromEnvironment();
```

Comment:
Yes, CommandContext.InitializeAnsiPassThroughFromEnvironment() reads the environmental variable (or uses the default value), while the following line sets the environmental variable true, which will effect any spawned processes.

As some background, all color formatting of output is done using [ASNI escape codes](https://github.com/dotnet/cli/blob/1376228e4ceade0e41e5723442672abdc17ea0e8/src/Microsoft.DotNet.Cli.Utils/AnsiColorExtensions.cs). However these are not supported by Windows so [Ansi Console](https://github.com/dotnet/cli/blob/a7f438f85d5a68c8f4c61bb422c39bbf3c1e1822/src/Microsoft.DotNet.Cli.Utils/AnsiConsole.cs#L66) parses them and changes Console.ForegroundColor appropriately.

The changes to Console.ForegroundColor have to be in the process that was directly invoked by the user, as the output redirection used when spawning processes does not capture the color changes. So the Report class [checks](https://github.com/dotnet/cli/blob/a7f438f85d5a68c8f4c61bb422c39bbf3c1e1822/src/Microsoft.DotNet.Cli.Utils/Reporter.cs#L35) the environmental variable DOTNET_CLI_CONTEXT_ANSI_PASS_THRU, and skips converting the ANSI codes.

Before the this pull request, the DOTNET_CLI_CONTEXT_ANSI_PASS_THRU was getting always set for intrinsic commands, so the ANSI escape codes were being written directly to the Windows console. This pull request makes the process use the current value of the environmental variable, which will default to false for the original invokcation of dotnet.exe. However we want any child process we spawn to pass through ANSI escape codes, so we need to set the environmental variable to true.